src/components/RegistrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,221 @@
+import { useState, FormEvent, ChangeEvent } from 'react';
+import { Link } from 'react-router-dom';
+
+import group_of_people from '../../assets/images/group_of_people.png';
+import { FormInput } from '../FormInput/FormInput';
+import style from './RegistrationForm.module.scss';
+
+const INITIAL_FORM_DATA = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  password: '',
+  confirmPassword: '',
+};
+
+const INITIAL_ERRORS = {
+  firstName: false,
+  lastName: false,
+  email: false,
+  password: false,
+  confirmPassword: false,
+};
+
+type FormData = typeof INITIAL_FORM_DATA;
+type FormErrors = typeof INITIAL_ERRORS;
+
+const ERROR_MESSAGES = {
+  firstName: 'First name cannot be empty or contain numbers',
+  lastName: 'Last name cannot be empty or contain numbers',
+  email: 'Please enter a valid email',
+  password:
+    'Password must have at least 8 characters, 1 uppercase letter and 1 number',
+  passwordMatch: "Passwords don't match",
+};
+
+const validators = {
+  name: (value: string): boolean => {
+    if (value.length === 0) {
+      return false;
+    }
+    for (let i = 0; i < value.length; i++) {
+      if (value[i] >= '0' && value[i] <= '9') {
+        return false;
+      }
+    }
+
+    return true;
+  },
+  email: (value: string): boolean => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value),
+  password: (value: string): boolean => {
+    let hasUpperCase = false;
+    let hasNumber = false;
+    if (value.length < 8) {
+      return false;
+    }
+    for (let i = 0; i < value.length; i++) {
```

Comment:
There is no need to use for loop, it is imperative, much more noisy (harder to scan through), easier to get wrong, and always requires mutation.

I cant think of any example, where declarative map/foreach/some/filter wouldn't suffice